packages/libraries/linkable-specs/src/__tests__/index.spec.ts

```diff
@@ -0,0 +1,79 @@
+import { DocumentNode, parse, StringValueNode, visit } from 'graphql';
+import { detectLinkedImplementations, LinkableSpec } from '../index';
+
+describe('index', () => {
+  test('LinkableSpec and detectLinkedImplementations can be used to easily implement linked schema functionality', () => {
+    const sdl = `
+          directive @meta(name: String!, content: String!) on SCHEMA | FIELD
+          directive @metadata_example(eg: String!) on FIELD
+          extend schema
+            @link(url: "https://specs.apollo.dev/link/v1.0")
+            @link(url: "https://specs.graphql-hive.com/metadata/v0.1", import: ["@meta"])
+  
+          type Query {
+            ping: String @meta(name: "owner", content: "hive-console-team")
+            pong: String @metadata__example(eg: "1...2...3... Pong")
+          }
+      `;
+
+    const metaSpec = new LinkableSpec('https://specs.graphql-hive.com/metadata', {
+      // The return value could be used to map sdl, collect information, or create a graphql yoga plugin.
+      // In this test, it's used to collect metadata information from the schema.
+      'v0.1': resolveImportName => (typeDefs: DocumentNode) => {
```

Comment:
This test is the best example of how I intended this API to be used.
The `LinkableSpec` defines how the linked schemas are implemented.
The `Link. resolveImportName ` function is then passed to the handler to be used to resolve the names of nodes as they are referred to within the subgraph... I chose to only pass `resolveImportName` to keep the API simpler.
And then the implementation uses that to figure out the names of things it supports.